builtin/providers/aws/resource_aws_s3_bucket.go

```diff
@@ -538,15 +538,18 @@ func resourceAwsS3BucketRead(d *schema.ResourceData, meta interface{}) error {
 		}
 	}
 
-	//read the acceleration status
-	accelerate, err := s3conn.GetBucketAccelerateConfiguration(&s3.GetBucketAccelerateConfigurationInput{
-		Bucket: aws.String(d.Id()),
-	})
-	log.Printf("[DEBUG] S3 bucket: %s, read Acceleration: %v", d.Id(), accelerate)
-	if err != nil {
-		return err
+	s3HostedRegion := meta.(*AWSClient).region
+	if s3HostedRegion != "cn-north-1" && s3HostedRegion != "us-gov-west-1" {
```

Comment:
I see what do you mean, the error message doesn't contain the name of unsupported argument - that's sad, true 😞 

It just depends on the level of reusability of that error code - the `isAWSErr` may potentially help us avoiding false positives/negatives when matching the error message. The question is do we want to ignore error messages we haven't seen yet in a context we didn't think about? IMO we don't because we don't know the exact effect of that. That's why more precise match _may_ be helpful.

The official API docs aren't very helpful in this area either. 😞  http://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html#ErrorCodeList doesn't contain `UnsupportedArgument` at all. Instead it contains bunch of 400 Bad Request errors with the **exact same code** related to transfer acceleration, most of which would _probably_ come from `POST`/Create operations.

I don't have a strong opinion about this specific case (`UnsupportedArgument` / `aws_s3_bucket`) though, either way is fine with me.